core/src/state.rs

```diff
@@ -20,17 +20,17 @@ pub struct State {
 }
 
 impl State {
-    pub fn new() -> Self {
+    pub fn new(context: Arc<Context>) -> Self {
         // @TODO file table
         // @see https://github.com/holochain/holochain-rust/pull/246
 
         let content_storage =
             MemoryStorage::new().expect("could not create new cas memory storage");
         let eav_storage = EavMemoryStorage::new().expect("could not create new eav memory storage");
-
+        let file_storage = &(*context).file_storage;
         State {
             nucleus: Arc::new(NucleusState::new()),
-            agent: Arc::new(AgentState::new(ChainStore::new(content_storage.clone()))),
+            agent: Arc::new(AgentState::new(ChainStore::new(file_storage.clone()))),
             dht: Arc::new(DhtStore::new(content_storage.clone(), eav_storage.clone())),
```

Comment:
Can we use the same storage for the DHT here as well? Is there a reason to leave DhtStore with the memory storage? @AshantiMutinta

Reply:
No you are right, just forgot the push  before I left. there in a few seconds.